pkg/controller/deployment/deployment_controller.go

```diff
@@ -557,9 +597,23 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment) (*api
 		dc.rcExpectations.DeleteExpectations(dKey)
 		return nil, fmt.Errorf("error creating replication controller: %v", err)
 	}
+
+	if err = dc.updateDeploymentVersion(deployment, newVersion); err != nil {
+		return createdRC, err
+	}
+
 	return createdRC, nil
 }
 
+func (dc *DeploymentController) updateDeploymentVersion(deployment extensions.Deployment, version string) error {
+	if deployment.Annotations == nil {
+		deployment.Annotations = make(map[string]string)
+	}
+	deployment.Annotations[deploymentutil.VersionAnnotation] = version
```

Comment:
We also need generation and `observedGeneration`, similar to RC.

I haven't looked at the RC implementation, but `observedGeneration` should be updated once the changed made in the corresponding `generation` have made to the underlying resources (RCs in this case).

But versions/revisions have nothing to do with that. This is about creating understandable history.